src/pudl/transform/ferc714.py

```diff
@@ -319,6 +340,270 @@ def _pre_process(df: pd.DataFrame, table_name: str) -> pd.DataFrame:
     return out_df
 
 
+def melt_hourx_columns_csv(df):
+    """Melt hourX columns into hours."""
+    # Almost all 25th hours are unusable (0.0 or daily totals),
+    # and they shouldn't really exist at all based on FERC instructions.
+    df = df.drop(columns="hour25")
+
+    # Melt daily rows with 24 demands to hourly rows with single demand
+    logger.info("Melting daily FERC 714 records into hourly records.")
+    df = df.rename(
+        columns=lambda x: int(re.sub(r"^hour", "", x)) - 1 if "hour" in x else x,
+    )
+    df = df.melt(
+        id_vars=[
+            "respondent_id_ferc714",
+            "report_year",
+            "report_date",
+            "utc_offset_code",
+        ],
+        value_vars=range(24),
+        var_name="hour",
+        value_name="demand_mwh",
+    )
+    return df
+
+
+def map_respondent_id_ferc714(df, source: Literal["csv", "xbrl"]):
+    """TODO: Make this map!"""
+    return df
+
+
+def remove_yearly_records(duration_xbrl):
+    """Convert a table with mostly daily records with some annuals into fully daily.
+
+    Almost all of the records have a start_date that == the end_date
+    which I'm assuming means the record spans the duration of one day
+    there are a small handful of records which seem to span a full year.
+    """
+    duration_xbrl = duration_xbrl.astype(
+        {"start_date": "datetime64[ns]", "end_date": "datetime64[ns]"}
+    )
+    one_day_mask = duration_xbrl.start_date == duration_xbrl.end_date
+    duration_xbrl_one_day = duration_xbrl[one_day_mask]
+    duration_xbrl_one_year = duration_xbrl[~one_day_mask]
+    # ensure there are really only a few of these multi-day records
+    assert len(duration_xbrl_one_year) / len(duration_xbrl_one_day) < 0.0005
+    # ensure all of these records are one year records
+    assert all(
+        duration_xbrl_one_year.start_date
+        + pd.DateOffset(years=1)
+        - pd.DateOffset(days=1)
+        == duration_xbrl_one_year.end_date
+    )
+    # these one-year records all show up as one-day records.
+    idx = ["entity_id", "report_year", "start_date"]
+    assert all(
+        duration_xbrl_one_year.merge(
+            duration_xbrl_one_day, on=idx, how="left", indicator=True
+        )._merge
+        == "both"
+    )
+    # all but two of them have the same timezone as the hourly data.
+    # two of them have UTC instead of a local timezone reported in hourly data.
+    # this leads me to think these are okay to just drop
+    return duration_xbrl_one_day
+
+
+def assign_report_day(df: pd.DataFrame, date_col: str) -> pd.DataFrame:
+    """Add a report_day column."""
+    return df.assign(
+        report_day=pd.to_datetime(df[date_col], format="%Y-%m-%d", exact=False)
+    )
+
+
+def merge_instant_and_duration_tables_xbrl(
```

Comment:
i almost used the ferc1 version of this but backed out and decided not to. that method calls other methods within the broader ferc table transformer class and it was hard to disentangle. also it was way more complex than this one needed to be. and also also this one needed to be done using different PKs and we needed to assign the day to merge on that. so i used the name and the general shape but not the code itself